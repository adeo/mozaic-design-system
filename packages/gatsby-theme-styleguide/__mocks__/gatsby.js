const React = require(`react`)
const gatsby = jest.requireActual(`gatsby`)

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
}
