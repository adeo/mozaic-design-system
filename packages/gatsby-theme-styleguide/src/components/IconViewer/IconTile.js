import React, { useState } from 'react'
import Copy from '../Copy'
import * as styles from './iconviewer.module.css'

const buildIconData = (name, icons) => {
  let dataIcon = {
    name: { name },
    currentSize: '48px',
    sizes: {
      '16px': {
        url: null,
        component: null,
        fileName: null,
        fullName: null,
        iconName: null,
      },
      '24px': {
        url: null,
        component: null,
        fileName: null,
        fullName: null,
        iconName: null,
      },
      '32px': {
        url: null,
        component: null,
        fileName: null,
        fullName: null,
        iconName: null,
      },
      '48px': {
        url: null,
        component: null,
        fileName: null,
        fullName: null,
        iconName: null,
      },
      '64px': {
        url: null,
        component: null,
        fileName: null,
        fullName: null,
        iconName: null,
      },
    },
  }

  Object.keys(dataIcon.sizes).forEach((size) => {
    if (!icons[size]) {
      return false
    } else {
      dataIcon.sizes[size].component = icons[size].componentName || null
      dataIcon.sizes[size].fileName = icons[size].fileName
      dataIcon.sizes[size].iconName = icons[size].iconName
    }

    if (!icons[dataIcon.currentSize]) {
      dataIcon.currentSize = Object.keys(icons).find((size) => {
        return icons[size] != null
      })
    }

    return true
  })

  return dataIcon
}

const createWebComponentName = (str) =>
  str.replace(/_/g, '-').replace('.svg', '').toLowerCase()

const TechnicalName = ({ keyName, value }) => (
  <p className={styles.technicalName}>
    {keyName}
    <br />
    {value}
  </p>
)

const IconTile = (props) => {
  const dataIcon = buildIconData(props.name, props.icons)
  const [currentSize, setCurrentSize] = useState(dataIcon.currentSize)
  const [allIcon, setAllIcon] = useState(dataIcon.sizes)

  function changeIconSize(size) {
    setCurrentSize(size)
  }

  return (
    <li className={styles.detail}>
      <p className={styles.iconName}>{props.name}</p>
      <div className={styles.iconWrapper}>
        {props.type === 'monochrom' ? (
          <img
            src={`/icons/monochrom/${allIcon[currentSize].fileName}`}
            alt={allIcon[currentSize].fileName}
            height={currentSize}
            width={currentSize}
            loading="lazy"
          />
        ) : (
          <img
            src={`/icons/color/${allIcon[currentSize].fileName}`}
            alt={allIcon[currentSize].fileName}
            height={currentSize}
            width={currentSize}
            loading="lazy"
          />
        )}
      </div>
      <ul className={styles.sizeList}>
        {Object.keys(dataIcon.sizes).map((size) => (
          <li
            key={`${props.name}-${size}`}
            className={`
              ${styles.sizeListItem} 
              ${currentSize === size ? styles.isActive : ''}
              ${!allIcon[size].fileName ? styles.isDisabled : ''}
            `}
          >
            <button
              type="button"
              className={styles.sizeButton}
              onClick={(e) => changeIconSize(size, e)}
              disabled={allIcon[size].fileName ? false : true}
            >
              {size}
            </button>
          </li>
        ))}
      </ul>
      {allIcon[currentSize].iconName && (
        <Copy
          value={allIcon[currentSize].iconName}
          children={
            <TechnicalName
              keyName="Name: "
              value={allIcon[currentSize].iconName}
            />
          }
        />
      )}
      {allIcon[currentSize].component && (
        <Copy
          value={allIcon[currentSize].component}
          children={
            <TechnicalName
              keyName="React/Svelte/Vue:"
              value={allIcon[currentSize].component}
            />
          }
        />
      )}
      {allIcon[currentSize].fileName && (
        <Copy
          value={createWebComponentName(allIcon[currentSize].fileName)}
          children={
            <TechnicalName
              keyName="Web Component: "
              value={createWebComponentName(allIcon[currentSize].fileName)}
            />
          }
        />
      )}
      <Copy
        value={allIcon[currentSize].fileName}
        children={
          <TechnicalName
            keyName="Svg file : "
            value={allIcon[currentSize].fileName}
          />
        }
      />
    </li>
  )
}

export default IconTile
