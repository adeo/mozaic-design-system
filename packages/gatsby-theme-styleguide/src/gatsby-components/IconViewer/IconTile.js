import React, { Component } from 'react'
import styled from 'styled-components'
import Copy from '../Copy'
import * as icons from '@mozaic-ds/icons/react'

const IconDetail = styled.li`
  flex: 0 1 30.8%;
  list-style: none;
  border: 1px solid #eee;
  box-sizing: border-box;
  border-radius: 4px;
  margin: 0.5rem;
`

const IconWrapper = styled.div`
  text-align: center;
  height: 104px;
  line-height: 3.75;
  display: flex;

  img,
  svg {
    margin: auto;
  }
`

const SizesList = styled.ul`
  display: flex;
  margin: 0 !important;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
`
const SizeItem = styled.li`
  flex: 0 1 20%;
  list-style: none;
  text-align: center;
  font-size: 0.65rem;
  line-height: 1.62rem;

  background: #f2f2f2;
  color: darkgrey;

  &:not(:first-child) {
    border-left: 1px solid #eee;
  }
`

const SizeItemAvail = styled(SizeItem)`
  background: ${({ isActive }) => (isActive ? `#78be20` : `none`)};
  color: ${({ isActive }) => (isActive ? `white !important` : `inherit`)};
`

export const SizeItemBtn = styled.button`
  box-shadow: none;
  margin: 0;
  padding: 0;
  border: none;
  font-size: 0.65rem;
  line-height: 1;
  background: none;
  cursor: pointer;
  color: inherit;
  padding: 0.5rem 0.25rem;
`

const IconName = styled.p`
  padding: 1rem;
  border-bottom: 1px solid #eee;
  font-size: 0.75rem;
  margin: 0;
  font-weight: 600;
  word-break: break-all;
  height: 40px;
`

const IconSizeName = styled.p`
  padding: 0 3px;
  margin: 3px;
  line-height: 1.25;
  display: block;
  word-break: break-all;
  position: relative;
`

export default class IconTile extends Component {
  constructor(props) {
    super(props)

    this.dataIcon = this.buildIconData(props.name, props.icons)

    this.state = {
      currentSize: this.dataIcon.currentSize,
      allIcon: this.dataIcon.sizes,
    }
  }

  changeIconSize(size) {
    this.setState({ currentSize: size })
  }

  buildIconData(name, icons) {
    let dataIcon = {
      name: { name },
      currentSize: '48px',
      sizes: {
        '16px': {
          url: null,
          component: null,
          fileName: null,
          fullName: null,
          iconName: null,
        },
        '24px': {
          url: null,
          component: null,
          fileName: null,
          fullName: null,
          iconName: null,
        },
        '32px': {
          url: null,
          component: null,
          fileName: null,
          fullName: null,
          iconName: null,
        },
        '48px': {
          url: null,
          component: null,
          fileName: null,
          fullName: null,
          iconName: null,
        },
        '64px': {
          url: null,
          component: null,
          fileName: null,
          fullName: null,
          iconName: null,
        },
      },
    }

    Object.keys(dataIcon.sizes).forEach(size => {
      if (!icons[size]) {
        return false
      } else {
        dataIcon.sizes[size].component = icons[size].componentName || null
        dataIcon.sizes[size].fileName = icons[size].fileName
        dataIcon.sizes[size].iconName = icons[size].iconName
      }

      if (!icons[dataIcon.currentSize]) {
        dataIcon.currentSize = Object.keys(icons).find(size => {
          return icons[size] != null
        })
      }

      return true
    })

    return dataIcon
  }

  render() {
    const Icon = icons[this.state.allIcon[this.state.currentSize].component]

    return (
      <IconDetail>
        <IconName>{this.props.name}</IconName>
        <IconWrapper>
          {this.props.type === 'monochrom' ? (
            <Icon fill="#454545" size={this.state.currentSize} />
          ) : (
            <img
              src={`/colorsvg/${
                this.state.allIcon[this.state.currentSize].fileName
              }`}
              width={this.state.currentSize}
              height={this.state.currentSize}
              alt={this.state.allIcon[this.state.currentSize].fileName}
            />
          )}
        </IconWrapper>
        <SizesList>
          {Object.keys(this.dataIcon.sizes).map(size =>
            this.state.allIcon[size].fileName ? (
              <SizeItemAvail
                key={`${this.props.name}-${size}`}
                isActive={this.state.currentSize === size}
              >
                <SizeItemBtn
                  type="button"
                  onClick={e => this.changeIconSize(size, e)}
                >
                  {size}
                </SizeItemBtn>
              </SizeItemAvail>
            ) : (
              <SizeItem key={`${this.props.name}-${size}`}>{size}</SizeItem>
            )
          )}
        </SizesList>
        {this.state.allIcon[this.state.currentSize].iconName && (
          <Copy
            value={this.state.allIcon[this.state.currentSize].iconName}
            children={
              <TechnicalName
                keyName="Name : "
                value={this.state.allIcon[this.state.currentSize].iconName}
              />
            }
          />
        )}
        {this.state.allIcon[this.state.currentSize].component && (
          <Copy
            value={this.state.allIcon[this.state.currentSize].component}
            children={
              <TechnicalName
                keyName="React/vue : "
                value={this.state.allIcon[this.state.currentSize].component}
              />
            }
          />
        )}
        <Copy
          value={this.state.allIcon[this.state.currentSize].fileName}
          children={
            <TechnicalName
              keyName="Svg file : "
              value={this.state.allIcon[this.state.currentSize].fileName}
            />
          }
        />
      </IconDetail>
    )
  }
}

const TechnicalName = ({ keyName, value }) => (
  <IconSizeName>
    {keyName}
    <br />
    {value}
  </IconSizeName>
)
