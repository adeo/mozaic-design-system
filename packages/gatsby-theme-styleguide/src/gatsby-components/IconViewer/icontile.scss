
.iconSize_16px {
  width: 16px;
  height: 16px;
}

.iconSize_24px {
  width: 24px;
  height: 24px;
}

.iconSize_32px {
  width: 32px;
  height: 32px;
}

.iconSize_48px {
  width: 48px;
  height: 48px;
}

.iconSize_64px {
  width: 64px;
  height: 64px;
}
