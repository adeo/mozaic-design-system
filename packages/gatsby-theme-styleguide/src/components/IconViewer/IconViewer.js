import React, { useState } from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import IconTile from './IconTile'
import * as styles from './iconviewer.module.css'

const IconViewer = ({ type }) => {
  const [filter, setFilter] = useState('')
  const data = useStaticQuery(graphql`
    query AllIcons {
      dataJson {
        monochrom {
          cat
          size
          name
          fileName
          iconName
          componentName
        }
      }
    }
  `)

  const icons = data.dataJson[type]
  const categories = {}

  icons.forEach((icon) => {
    if (!categories[icon.cat]) categories[icon.cat] = {}

    if (!categories[icon.cat][icon.name])
      categories[icon.cat][icon.name] = {
        20: null,
        24: null,
        32: null,
        48: null,
        64: null,
      }

    if (categories[icon.cat][icon.name][icon.size] === null)
      categories[icon.cat][icon.name][icon.size] = {
        ...icon,
      }

    return null
  })

  return (
    <div>
      <input
        id="filter"
        type="text"
        value={filter}
        className="mc-text-input"
        placeholder="Search icons"
        onChange={(event) => setFilter(event.target.value)}
      />
      {Object.keys(categories)
        .filter((cat) => {
          return cat.length > 0
        })
        .map((cat) => {
          return (
            <div key={cat}>
              {Object.keys(categories[cat]).filter(
                (f) => f.includes(filter) || filter === '',
              ).length > 0 && <h2 className={styles.subtitle}>{cat}</h2>}
              <ul className={styles.list}>
                {Object.keys(categories[cat])
                  .filter((f) => f.includes(filter) || filter === '')
                  .map((name) => (
                    <IconTile
                      type={type}
                      key={`${cat}-${name}`}
                      name={name}
                      icons={categories[cat][name]}
                    />
                  ))}
              </ul>
            </div>
          )
        })}
    </div>
  )
}

export default IconViewer
