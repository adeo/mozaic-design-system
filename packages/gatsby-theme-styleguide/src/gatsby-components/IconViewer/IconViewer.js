import styled from 'styled-components'
import React from 'react'
import { StaticQuery, graphql } from 'gatsby'

import IconTile from './IconTile'

const IconsList = styled.ul`
  padding: 0;
  margin: 0 !important;
  display: flex;
  flex-wrap: wrap;
`

const IconViewer = ({ type }) => {
  return (
    <StaticQuery
      query={graphql`
        query AllIcons {
          dataJson {
            monochrom {
              cat
              componentName
              fileName
              name
              iconName
              size
            }
            color {
              cat
              fileName
              name
              size
            }
          }
        }
      `}
      render={data => {
        const icons = data.dataJson[type]
        const categories = {}

        icons.forEach(icon => {
          if (!categories[icon.cat]) categories[icon.cat] = {}

          if (!categories[icon.cat][icon.name])
            categories[icon.cat][icon.name] = {
              '16px': null,
              '24px': null,
              '32px': null,
              '48px': null,
              '64px': null,
            }

          if (categories[icon.cat][icon.name][icon.size] === null)
            categories[icon.cat][icon.name][icon.size] = {
              ...icon,
            }

          return null
        })

        return (
          <div>
            {Object.keys(categories).map(cat => (
              <div key={cat}>
                <h1>{cat}</h1>
                <IconsList>
                  {Object.keys(categories[cat]).map(name => (
                    <IconTile
                      type={type}
                      key={`${cat}-${name}`}
                      name={name}
                      icons={categories[cat][name]}
                    />
                  ))}
                </IconsList>
              </div>
            ))}
          </div>
        )
      }}
    />
  )
}

export default IconViewer
