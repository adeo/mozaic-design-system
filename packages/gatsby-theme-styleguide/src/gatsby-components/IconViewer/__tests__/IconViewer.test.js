import React from 'react'
import renderer from 'react-test-renderer'
import { StaticQuery, query } from 'gatsby'

import mock from './IconViewer.mock'
import IconViewer from '../IconViewer'

beforeEach(() => {
  StaticQuery.mockImplementation(({ render }) => render(mock))
})

describe('<IconViewer />', () => {
  it('render corectly monochrom icons', () => {
    const tree = renderer.create(<IconViewer type="monochrom" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render corectly colors icon', () => {
    const tree = renderer.create(<IconViewer type="color" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
