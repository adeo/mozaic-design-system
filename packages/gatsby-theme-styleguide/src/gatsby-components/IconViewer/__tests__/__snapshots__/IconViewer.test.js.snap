// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconViewer /> render corectly colors icon 1`] = `
.c8 {
  width: 100%;
  padding: 3px 5px;
  font-size: 10px;
  color: #999;
  line-height: 2em;
  cursor: pointer;
  position: relative;
  box-sizing: border-box;
}

.c8:not(:last-child) {
  border-bottom: solid 1px #f5f5f5;
}

.c1 {
  -webkit-flex: 0 1 30.8%;
  -ms-flex: 0 1 30.8%;
  flex: 0 1 30.8%;
  list-style: none;
  border: 1px solid #eee;
  box-sizing: border-box;
  border-radius: 4px;
  margin: 0.5rem;
}

.c3 {
  text-align: center;
  height: 104px;
  line-height: 3.75;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 img,
.c3 svg {
  margin: auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 !important;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}

.c5 {
  -webkit-flex: 0 1 20%;
  -ms-flex: 0 1 20%;
  flex: 0 1 20%;
  list-style: none;
  text-align: center;
  font-size: 0.65rem;
  line-height: 1.62rem;
  background: #f2f2f2;
  color: darkgrey;
}

.c5:not(:first-child) {
  border-left: 1px solid #eee;
}

.c6 {
  -webkit-flex: 0 1 20%;
  -ms-flex: 0 1 20%;
  flex: 0 1 20%;
  list-style: none;
  text-align: center;
  font-size: 0.65rem;
  line-height: 1.62rem;
  background: #f2f2f2;
  color: darkgrey;
  background: #78be20;
  color: white !important;
}

.c6:not(:first-child) {
  border-left: 1px solid #eee;
}

.c7 {
  box-shadow: none;
  margin: 0;
  padding: 0;
  border: none;
  font-size: 0.65rem;
  line-height: 1;
  background: none;
  cursor: pointer;
  color: inherit;
  padding: 0.5rem 0.25rem;
}

.c2 {
  padding: 1rem;
  border-bottom: 1px solid #eee;
  font-size: 0.75rem;
  margin: 0;
  font-weight: 600;
  word-break: break-all;
  height: 40px;
}

.c9 {
  padding: 0 3px;
  margin: 3px;
  line-height: 1.25;
  display: block;
  word-break: break-all;
  position: relative;
}

.c0 {
  padding: 0;
  margin: 0 !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <div>
    <h1>
      Payment
    </h1>
    <ul
      className="c0"
    >
      <li
        className="c1"
      >
        <p
          className="c2"
        >
          Cards_Colors_Mastercard
        </p>
        <div
          className="c3"
        >
          <img
            alt="Payment_Cards_Colors_Mastercard_32px.svg"
            height="32px"
            src="/colorsvg/Payment_Cards_Colors_Mastercard_32px.svg"
            width="32px"
          />
        </div>
        <ul
          className="c4"
        >
          <li
            className="c5"
          >
            16px
          </li>
          <li
            className="c5"
          >
            24px
          </li>
          <li
            className="c6"
          >
            <button
              className="c7"
              onClick={[Function]}
              type="button"
            >
              32px
            </button>
          </li>
          <li
            className="c5"
          >
            48px
          </li>
          <li
            className="c5"
          >
            64px
          </li>
        </ul>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            Svg file : 
            <br />
            Payment_Cards_Colors_Mastercard_32px.svg
          </p>
        </div>
      </li>
      <li
        className="c1"
      >
        <p
          className="c2"
        >
          Cards_Colors_Paylib
        </p>
        <div
          className="c3"
        >
          <img
            alt="Payment_Cards_Colors_Paylib_64px.svg"
            height="64px"
            src="/colorsvg/Payment_Cards_Colors_Paylib_64px.svg"
            width="64px"
          />
        </div>
        <ul
          className="c4"
        >
          <li
            className="c5"
          >
            16px
          </li>
          <li
            className="c5"
          >
            24px
          </li>
          <li
            className="c5"
          >
            32px
          </li>
          <li
            className="c5"
          >
            48px
          </li>
          <li
            className="c6"
          >
            <button
              className="c7"
              onClick={[Function]}
              type="button"
            >
              64px
            </button>
          </li>
        </ul>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            Svg file : 
            <br />
            Payment_Cards_Colors_Paylib_64px.svg
          </p>
        </div>
      </li>
      <li
        className="c1"
      >
        <p
          className="c2"
        >
          Cards_Colors_Paypal
        </p>
        <div
          className="c3"
        >
          <img
            alt="Payment_Cards_Colors_Paypal_64px.svg"
            height="64px"
            src="/colorsvg/Payment_Cards_Colors_Paypal_64px.svg"
            width="64px"
          />
        </div>
        <ul
          className="c4"
        >
          <li
            className="c5"
          >
            16px
          </li>
          <li
            className="c5"
          >
            24px
          </li>
          <li
            className="c5"
          >
            32px
          </li>
          <li
            className="c5"
          >
            48px
          </li>
          <li
            className="c6"
          >
            <button
              className="c7"
              onClick={[Function]}
              type="button"
            >
              64px
            </button>
          </li>
        </ul>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            Svg file : 
            <br />
            Payment_Cards_Colors_Paypal_64px.svg
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<IconViewer /> render corectly monochrom icons 1`] = `
.c8 {
  width: 100%;
  padding: 3px 5px;
  font-size: 10px;
  color: #999;
  line-height: 2em;
  cursor: pointer;
  position: relative;
  box-sizing: border-box;
}

.c8:not(:last-child) {
  border-bottom: solid 1px #f5f5f5;
}

.c1 {
  -webkit-flex: 0 1 30.8%;
  -ms-flex: 0 1 30.8%;
  flex: 0 1 30.8%;
  list-style: none;
  border: 1px solid #eee;
  box-sizing: border-box;
  border-radius: 4px;
  margin: 0.5rem;
}

.c3 {
  text-align: center;
  height: 104px;
  line-height: 3.75;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 img,
.c3 svg {
  margin: auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 !important;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}

.c7 {
  -webkit-flex: 0 1 20%;
  -ms-flex: 0 1 20%;
  flex: 0 1 20%;
  list-style: none;
  text-align: center;
  font-size: 0.65rem;
  line-height: 1.62rem;
  background: #f2f2f2;
  color: darkgrey;
}

.c7:not(:first-child) {
  border-left: 1px solid #eee;
}

.c5 {
  -webkit-flex: 0 1 20%;
  -ms-flex: 0 1 20%;
  flex: 0 1 20%;
  list-style: none;
  text-align: center;
  font-size: 0.65rem;
  line-height: 1.62rem;
  background: #f2f2f2;
  color: darkgrey;
  background: #78be20;
  color: white !important;
}

.c5:not(:first-child) {
  border-left: 1px solid #eee;
}

.c6 {
  box-shadow: none;
  margin: 0;
  padding: 0;
  border: none;
  font-size: 0.65rem;
  line-height: 1;
  background: none;
  cursor: pointer;
  color: inherit;
  padding: 0.5rem 0.25rem;
}

.c2 {
  padding: 1rem;
  border-bottom: 1px solid #eee;
  font-size: 0.75rem;
  margin: 0;
  font-weight: 600;
  word-break: break-all;
  height: 40px;
}

.c9 {
  padding: 0 3px;
  margin: 3px;
  line-height: 1.25;
  display: block;
  word-break: break-all;
  position: relative;
}

.c0 {
  padding: 0;
  margin: 0 !important;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <div>
    <h1>
      Media
    </h1>
    <ul
      className="c0"
    >
      <li
        className="c1"
      >
        <p
          className="c2"
        >
          FullScreen
        </p>
        <div
          className="c3"
        >
          <svg
            fill="#454545"
            height="16px"
            viewBox="0 0 16 16"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.5 14H3.25A1.25 1.25 0 012 12.75V10.5a1 1 0 012 0V12h1.5a1 1 0 010 2zM3 6.5a1 1 0 01-1-1V3.25A1.25 1.25 0 013.25 2H5.5a1 1 0 010 2H4v1.5a1 1 0 01-1 1zM13 6.5a1 1 0 01-1-1V4h-1.5a1 1 0 010-2h2.25A1.25 1.25 0 0114 3.25V5.5a1 1 0 01-1 1zM12.75 4zM12.75 14H10.5a1 1 0 010-2H12v-1.5a1 1 0 012 0v2.25A1.25 1.25 0 0112.75 14zM9.75 11h-3.5A1.25 1.25 0 015 9.75v-3.5A1.25 1.25 0 016.25 5h3.5A1.25 1.25 0 0111 6.25v3.5A1.25 1.25 0 019.75 11zM7 9h2V7H7z"
            />
          </svg>
        </div>
        <ul
          className="c4"
        >
          <li
            className="c5"
          >
            <button
              className="c6"
              onClick={[Function]}
              type="button"
            >
              16px
            </button>
          </li>
          <li
            className="c7"
          >
            24px
          </li>
          <li
            className="c7"
          >
            32px
          </li>
          <li
            className="c7"
          >
            48px
          </li>
          <li
            className="c7"
          >
            64px
          </li>
        </ul>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            React/vue : 
            <br />
            FullScreen16
          </p>
        </div>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            Svg file : 
            <br />
            Media_FullScreen_16px.svg
          </p>
        </div>
      </li>
      <li
        className="c1"
      >
        <p
          className="c2"
        >
          ReduceScreen
        </p>
        <div
          className="c3"
        >
          <svg
            fill="#454545"
            height="16px"
            viewBox="0 0 16 16"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.7 9H3a1 1 0 000 2h2v2a1 1 0 002 0v-2.7A1.3 1.3 0 005.7 9zM6 2a1 1 0 00-1 1v2H3a1 1 0 000 2h2.7A1.3 1.3 0 007 5.7V3a1 1 0 00-1-1zM10.3 7H13a1 1 0 000-2h-2V3a1 1 0 00-2 0v2.7A1.3 1.3 0 0010.3 7zM13 9h-2.7A1.3 1.3 0 009 10.3V13a1 1 0 002 0v-2h2a1 1 0 000-2z"
            />
          </svg>
        </div>
        <ul
          className="c4"
        >
          <li
            className="c5"
          >
            <button
              className="c6"
              onClick={[Function]}
              type="button"
            >
              16px
            </button>
          </li>
          <li
            className="c7"
          >
            24px
          </li>
          <li
            className="c7"
          >
            32px
          </li>
          <li
            className="c7"
          >
            48px
          </li>
          <li
            className="c7"
          >
            64px
          </li>
        </ul>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            React/vue : 
            <br />
            ReduceScreen16
          </p>
        </div>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            Svg file : 
            <br />
            Media_ReduceScreen_16px.svg
          </p>
        </div>
      </li>
    </ul>
  </div>
  <div>
    <h1>
      Social
    </h1>
    <ul
      className="c0"
    >
      <li
        className="c1"
      >
        <p
          className="c2"
        >
          SocialNetwork_Linkedin
        </p>
        <div
          className="c3"
        >
          <svg
            fill="#454545"
            height="48px"
            viewBox="0 0 48 48"
            width="48px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M25.28 22.35v-.06.06zM39 7H9a2 2 0 00-2 2v30a2 2 0 002 2h30a2 2 0 002-2V9a2 2 0 00-2-2zM17.36 35.48h-5.12V20.12h5.12zm-.47-18.21a2.9 2.9 0 01-2.1.77 2.82 2.82 0 01-2-.77 2.55 2.55 0 01-.79-1.89 2.47 2.47 0 01.81-1.9 2.92 2.92 0 012.09-.76 2.79 2.79 0 012.06.76 2.57 2.57 0 01.8 1.9 2.52 2.52 0 01-.87 1.89zm18.86 18.21h-5.11v-8.21q0-3.48-2.57-3.48a2.53 2.53 0 00-1.64.54 3.46 3.46 0 00-1 1.32 3.5 3.5 0 00-.16 1.24v8.59h-5.1q.06-13.9 0-15.36h5.11v2.17a5.15 5.15 0 014.58-2.53 5.58 5.58 0 014.28 1.75 7.4 7.4 0 011.61 5.16zM25.28 22.35v-.06.06z"
            />
            <path
              d="M39 7H9a2 2 0 00-2 2v30a2 2 0 002 2h30a2 2 0 002-2V9a2 2 0 00-2-2zM17.36 35.48h-5.12V20.12h5.12zm-.47-18.21a2.9 2.9 0 01-2.1.77 2.82 2.82 0 01-2-.77 2.55 2.55 0 01-.79-1.89 2.47 2.47 0 01.81-1.9 2.92 2.92 0 012.09-.76 2.79 2.79 0 012.06.76 2.57 2.57 0 01.8 1.9 2.52 2.52 0 01-.87 1.89zm18.86 18.21h-5.11v-8.21q0-3.48-2.57-3.48a2.53 2.53 0 00-1.64.54 3.46 3.46 0 00-1 1.32 3.5 3.5 0 00-.16 1.24v8.59h-5.1q.06-13.9 0-15.36h5.11v2.17a5.15 5.15 0 014.58-2.53 5.58 5.58 0 014.28 1.75 7.4 7.4 0 011.61 5.16z"
            />
          </svg>
        </div>
        <ul
          className="c4"
        >
          <li
            className="c7"
          >
            16px
          </li>
          <li
            className="c7"
          >
            24px
          </li>
          <li
            className="c7"
          >
            32px
          </li>
          <li
            className="c5"
          >
            <button
              className="c6"
              onClick={[Function]}
              type="button"
            >
              48px
            </button>
          </li>
          <li
            className="c7"
          >
            64px
          </li>
        </ul>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            React/vue : 
            <br />
            SocialNetworkLinkedin48
          </p>
        </div>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            Svg file : 
            <br />
            Social_SocialNetwork_Linkedin_48px.svg
          </p>
        </div>
      </li>
      <li
        className="c1"
      >
        <p
          className="c2"
        >
          SocialNetwork_Pinterest
        </p>
        <div
          className="c3"
        >
          <svg
            fill="#454545"
            height="48px"
            viewBox="0 0 48 48"
            width="48px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M39 7H9a2 2 0 00-2 2v30a2 2 0 002 2h8.42a16.86 16.86 0 01.16-5.62l2.17-9.17a6 6 0 01-.53-2.68 5 5 0 01.94-3.08 2.79 2.79 0 012.29-1.26 2 2 0 011.67.72 2.79 2.79 0 01.59 1.81 6.55 6.55 0 01-.25 1.65c-.16.64-.38 1.39-.65 2.26s-.47 1.54-.59 2a2.51 2.51 0 00.51 2.37 2.67 2.67 0 002.15 1q2.34 0 3.84-2.62A12.51 12.51 0 0031.21 22a6.17 6.17 0 00-1.84-4.64 7.08 7.08 0 00-5.15-1.8 8 8 0 00-6 2.36 7.87 7.87 0 00-2.29 5.68 5.12 5.12 0 001.11 3.32 1 1 0 01.24.93q-.11.38-.33 1.35a.59.59 0 01-.84.42 5 5 0 01-2.58-2.44 8.74 8.74 0 01-.88-4 9.41 9.41 0 01.48-2.93 10.26 10.26 0 011.47-2.84A12.39 12.39 0 0117 15a10.76 10.76 0 013.39-1.69 13.93 13.93 0 014.3-.64 11.63 11.63 0 014.37.81 10.21 10.21 0 013.37 2.15 10 10 0 012.14 3.08 8.81 8.81 0 01.77 3.59 13.08 13.08 0 01-2.53 8.27 7.91 7.91 0 01-6.4 3.27 5.4 5.4 0 01-2.47-.54 3.9 3.9 0 01-1.79-1.57c-.65 2.59-1 4.13-1.17 4.63a15.56 15.56 0 01-2.4 4.64H39a2 2 0 002-2V9a2 2 0 00-2-2z"
            />
          </svg>
        </div>
        <ul
          className="c4"
        >
          <li
            className="c7"
          >
            16px
          </li>
          <li
            className="c7"
          >
            24px
          </li>
          <li
            className="c7"
          >
            32px
          </li>
          <li
            className="c5"
          >
            <button
              className="c6"
              onClick={[Function]}
              type="button"
            >
              48px
            </button>
          </li>
          <li
            className="c7"
          >
            64px
          </li>
        </ul>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            React/vue : 
            <br />
            SocialNetworkPinterest48
          </p>
        </div>
        <div
          className="c8"
          onClick={[Function]}
        >
          <p
            className="c9"
          >
            Svg file : 
            <br />
            Social_SocialNetwork_Pinterest_48px.svg
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
